src/main/java/com/google/cloud/tools/appengine/AppEngineDescriptor.java

```
@@ -99,24 +105,88 @@ public boolean isJava8() {
+   * </env-variables>
+   * }
+   * </pre>
+   * This will construct a map of the form {[key, val], ...}
+   *
+   * @return a map representing the environment variable settings in the appengine-web.xml.
+   */
+  public Map<String, String> getEnvironment() {
+    Node environmentParentNode = getNode(document, "appengine-web-app", "env-variables");
+    return getAttributeMap(environmentParentNode, "env-var", "name", "value");
+  }
+
+  private static String getText(Node node) {
+    if (node != null) {
+      try {
+        return node.getTextContent();
+      } catch (DOMException ex) {
+        // this shouldn't happen barring a very funky DOM implementation
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * @return a map formed from the attributes of the nodes contained within the parent node.
+   */
+  private static Map<String, String> getAttributeMap(Node parent,
+                                                     String nodeName,
+                                                     String keyAttributeName,
+                                                     String valueAttributeName) {
+    if (parent != null) {
+      Map<String, String> nameValueAttributeMap = Maps.newHashMap();
+
+      if (parent.hasChildNodes()) {
+        for (int i = 0; i < parent.getChildNodes().getLength(); ++i) {
+          Node child = parent.getChildNodes().item(i);
+          NamedNodeMap attributeMap = child.getAttributes();
+
+          if (nodeName.equals(child.getNodeName()) && attributeMap != null) {
+            Node keyNode = attributeMap.getNamedItem(keyAttributeName);
+
+            if (keyNode != null) {
+              Node valueNode = attributeMap.getNamedItem(valueAttributeName);
+              try {
+                nameValueAttributeMap.put(keyNode.getNodeValue(), valueNode.getNodeValue());
+              } catch (DOMException ex) {
+                // this shouldn't happen barring a very funky DOM implementation
```

swallowing exceptions seems risky.